### ISSUE TITLE
Follow ns alias to define functions

### ISSUE THREAD

**Issue by wilkerlucio** [kkinnear/zprint#77]
Hello, I just started configuring my zprint, but I noticed that when we override functions we do so by using the alias name, but different projects might use different aliases for a fn, but the formatting should stay the same. Would be possible to also use fully qualified names and have zprint auto-resolve those by reading the `ns` form?

**Comment by dpassen** [comment]
Not sure of the project etiquette but would like to bump this. This is a request my team has desired.

**Comment by kkinnear** [comment]
Yes, I seem to have lost this particular request in the shuffle of other work.  

At present, while everything is parsed, zprint doesn't semantically interpret almost any information other than `;!zprint ...` comments.  Not that it couldn't, of course.

As I think this through, if you were using zprint to format individual functions while in an editor, this wouldn't help you much, because zprint wouldn't see the `(ns ...)` declaration for that function.  But if you are just using zprint on whole files, I suppose that it could still work.

In thinking through how this might actually be implemented, I imagine that there might be an alias map in the options map, and that the alias map would get populated from the `(ns ...)` declaration.  

In looking in detail at the `ns` documentation, I'm seeing ways to alias the namespaces, but that isn't an issue, since zprint will trim off any namespace and look for just the function name in the `:fn-map`.  I am seeing this (in someone's "cheat sheet" for the `ns` macro):
```
(ns project.handler.addevent
    (:require 
        [project.view.pages :refer [addevent] :rename {addevent addevent-page}))
```
Is this what you are thinking -- have zprint look at the `(:require [ ... :rename {}]}` maps?  Or is there something I'm missing here?

I suppose that if you gave me a specific example of what you are looking for -- that is, an `ns` declaration and what you would want zprint to figure out about it, I would have a better chance of actually doing what you want.

I'm still not entirely committed to doing this, but I'm certainly willing explore it and there is a decent chance that I could pull this off in a relatively clean manner (which would make it more attractive to implement).

**Comment by dpassen** [comment]
Not speaking for @wilkerlucio , but my team wanted all clojure.test/testing blocks to have :arg1 entries (among other things), though some might refer the symbol, some may require it as test, and others as t. We have considered using clj-kondo to enforce [Consistent-alias](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#consistent-alias)

**Comment by kkinnear** [comment]
That helps, but just so I don't miss something, could you give me a concrete example of all of the three ways people might do one specific symbol?  I'm know that this is a bit of work.  Otherwise I'm trying to guess what you actually want from your description.  Which I might or might not get right.  Thanks!

**Comment by dpassen** [comment]
Sure.

```clj
(:require
  [clojure.test :refer [deftest is testing]])
```

```clj
(:require
  [clojure.test :as test]])
```

```clj
(:require
  [clojure.test :as t]])
```

Given that mess of requires, how best to automatically format `clojure.test/testing` the same way? And to not affect any other functions that may be named `testing`?

**Comment by kkinnear** [comment]
Thanks, that helps!

So, zprint currently looks up the entire string of a function (including the namespace and '/' if any) in the `:fn-map`.  If it doesn't find it, it trims off everything up to and including the '/', and then looks that up in the `:fn-map`.  So, if you just put `"testing"` in the `:fn-map`, it will always format the same regardless of the namespace on the front (or without a namespace on the front).

The only thing missing is the "And to not affect any other functions that may be named testing."  Off the top of my head, I don't see any way to fulfill that desire, given the first of the requires you gave:
```
(:require
  [clojure.test :refer [deftest is testing]])
```
since this will be referencing "testing" without any namespaces.  If there were other instances of `"testing"` that were not from `clojure.test`, then wouldn't this whole `:require` be in error?  Ah, yes, in this file it would.  But I suppose you are thinking that you would like zprint to look up `"clojure.test/testing"` in the `:fn-map`, and not `"testing"` in the `:fn-map` in this case.   Well, probably in all cases. 

So, what I'm taking away from this is that you would like to put this in the `:fn-map`:
```
{:fn-map {"clojure.test/testing" :arg1}}
```
and have it work for all of the above requires. In the first case, `testing` would match that entry, the second case `test/testing` would match that entry, and in the third case `t/testing` would match that entry.

Is that correct?

**Comment by dpassen** [comment]
Yes, exactly. It's very close to desired behavior. But consider `compojure.core/GET`. Would an entry of `"GET"` in the fn-map affect `clojure.core/get`?

**Comment by kkinnear** [comment]
Well, no, as the `:fn-map` is map of strings which are case sensitive.  But I see your point.

**Comment by dpassen** [comment]
I would have liked to have a better example than GET. But I'm glad it carried over
